### PR TITLE
chore: add metric integration tests for streaming updates

### PR DESCRIPTION
**Pull request by corban-beaird**
## Description
Addes metric integration tests, refactors test_util to account for metrics not including a knownSet, refactors streamable types to no longer export workspace_id.

## Test Plan
- none, run current integration tests

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-10033

### CODE REVIEW

**Comment by corban-beaird** [comment]
Looks like this isn't ready just yet, there's a parsing issue as a result of these changes a handful of e2e tests, working to resolve now

File: master/internal/stream/checkpoints.go [hunk 1]
```diff
@@ -60,9 +60,12 @@ func (c *CheckpointMsg) SeqNum() int64 {
 
 // UpsertMsg creates a Checkpoint upserted prepared message.
 func (c *CheckpointMsg) UpsertMsg() stream.UpsertMsg {
+	// omit workspaceID since it wasn't part of the original row
+	exportedMsg := *c
+	exportedMsg.WorkspaceID = 0
```
Comment (eecsliu) [inline master/internal/stream/checkpoints.go:65]: beautiful, thanks for updating!

File: master/internal/stream/system_intg_test.go [hunk 2]
```diff
@@ -465,6 +507,160 @@ func TestCheckpointStartup(t *testing.T) {
 	runStartupTest(t, testCases)
 }
 
+func TestMetricInsert(t *testing.T) {
+	pgDB, dbCleanup := db.MustResolveNewPostgresDatabase(t)
+	t.Cleanup(dbCleanup)
+
+	type metric struct {
```
Comment (eecsliu) [inline master/internal/stream/system_intg_test.go:514]: non-blocking: my only thought here is that if we need to test metrics in the future, wed want to move this out of this function. Im ok with this as it stands though and we can leave it to a future refactor

**Review by eecsliu — APPROVED** [review]
Looks good to me!

File: master/internal/stream/system_intg_test.go (hunk 2 above)
Comment (corban-beaird) [reply master/internal/stream/system_intg_test.go:514]: Fair point! It'll be a good addition to `test_util.go`

**Pull request merged**